### PR TITLE
221 manual passphrase entry

### PR DESCRIPTION
**Pull request by jakegraves-figure**
Changed gears in the middle of this branch, and started working with the `Autocomplete` widget.

`Autocomplete` was a bit different than what I was expecting, not letting me supply a `TextEditingController` but it gave me access to the one it had, so I made it work with the existing logic.

Testing with iOS and the Android emulator I was running, there didn't seem to be any bug regression.

### CODE REVIEW

**Review by royd — APPROVED** [review]

**Review by dhancockFigure — COMMENTED** [review]
Is there a way to make the suggestion popup stand out a little more? I missed a couple of times

**Comment by jakegraves-figure** [comment]
> Is there a way to make the suggestion popup stand out a little more? I missed a couple of times

I could probably make the suggestion a shade lighter.

**Comment by dhancockFigure** [comment]
> 
or even just add a border

**Review by dhancockFigure — APPROVED** [review]
looks good

File: lib/screens/recover_passphrase_entry_screen/recover_passphrase_bloc.dart [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+    if (textControllers[index] == controller) {
+      return;
+    } else {
+      textControllers[index] = controller;
+      _addListener(controller, index);
+    }
+  }
+
+  bool isMnemonicComplete() {
+    return textControllers.entries.length == 24 &&
```
Comment (royd) [inline lib/screens/recover_passphrase_entry_screen/recover_passphrase_bloc.dart:31]: Could use `_wordsCount`

File: lib/screens/recover_passphrase_entry_screen/recover_passphrase_entry_screen.dart [hunk 2]
```diff
@@ -166,11 +154,8 @@ class RecoverPassphraseEntryScreenState
             child: Column(
               key: RecoverPassphraseEntryScreen.wordList,
               mainAxisSize: MainAxisSize.min,
-              children: textControllers.map(
-                (textController) {
-                  final index = textControllers.indexOf(textController);
-                  final focusNode = focusNodes[index];
-
+              children: Iterable<int>.generate(24).toList().map(
```
Comment (royd) [inline lib/screens/recover_passphrase_entry_screen/recover_passphrase_entry_screen.dart:157]: For all the `23` and `24` values, I would base them off of `wordCount` from the bloc

**Review by royd — CHANGES_REQUESTED** [review]

**Review by royd — APPROVED** [review]

**Review by dhancockFigure — APPROVED** [review]

**Pull request merged**